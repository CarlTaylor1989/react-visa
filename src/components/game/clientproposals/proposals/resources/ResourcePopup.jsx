import React from 'react';
import PropTypes from 'prop-types';
import { Translate } from 'react-redux-i18n';
import ScrollBar from 'react-scrollbars-custom';
import scrollBarSettings from '../../../../../config/scrollBar';
import ResourceIcon from './ResourceIcon';
import './ResourcePopup.scss';
import ResouceContent from './ResourceContent';

const ResourcePopup = ({
  showPopup,
  client,
  closePopup,
  resourceId,
  type,
  resources
}) => {
  const [resourceIndex] = Object.keys(resources).filter(
    r => `res${parseInt(r, 10) + 1}` === resourceId
  );

  const resource = resources[resourceIndex];
  const extraClasses = showPopup ? ' available' : '';

  return (
    <div className={`resourcePopup${extraClasses}`}>
      <div className="popupWrapper">
        <div className="buttons">
          <ResourceIcon type={type} />
          <button
            type="button"
            className="exitPopup"
            onClick={closePopup}
          >
            <Translate value="generic.popupCloseBtn" />
          </button>
        </div>
        <div className="scrollingArea">
          <ScrollBar {...scrollBarSettings}>
            <ResouceContent
              client={client}
              resource={resource}
              resourceIndex={resourceIndex}
              type={type}
            />
          </ScrollBar>
        </div>
      </div>
    </div>
  );
};

ResourcePopup.propTypes = {
  client: PropTypes.string.isRequired,
  closePopup: PropTypes.func.isRequired,
  showPopup: PropTypes.bool.isRequired,
  resourceId: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  resources: PropTypes.object.isRequired
};

export default ResourcePopup;
