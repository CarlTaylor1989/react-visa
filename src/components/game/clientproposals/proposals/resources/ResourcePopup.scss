$popup-close-n: '../../../../../styles/icons/popup_close_n.svg';
$popup-close-h: '../../../../../styles/icons/popup_close_h.svg';

@import '../../../../../styles/generic';
@import '../../../../../styles/popup';

$audio-icon: '../../../../../styles/icons/clients/resource_icons/audio_icon.svg';
$call-icon: '../../../../../styles/icons/clients/resource_icons/call_icon.svg';
$email-icon: '../../../../../styles/icons/clients/resource_icons/email_icon.svg';
$pdf-icon: '../../../../../styles/icons/clients/resource_icons/pdf_icon.svg';
$video-icon: '../../../../../styles/icons/clients/resource_icons/video_icon.svg';
$icon-width: 54px;
$icon-height: 40px;

$avatar-icon: '../../../../../styles/icons/clients/resource_icons/resource_avatar_02.svg';
$avatar-icon-width: 100px;
$avatar-icon-height: $avatar-icon-width;

.resourcePopup {
  @extend %gradient-popup;
  @include border-radius(27px);
  @include box-shadow(21px, 21px, 11px, 0, rgba(0, 0, 0, .3));
  @include opacity(0);
  border: 2px solid $colour-white;
  bottom: -2px;
  box-sizing: border-box;
  left: -2px;
  position: absolute;
  right: -2px;
  top: -2px;
  transition: opacity .5s ease-in-out;
  z-index: -1;

  &.available {
    @include opacity(1);
    z-index: 10;
  }

  .popupWrapper {
    padding: 25px 25px 40px 50px;

    .resourceIcon {
      background-position: top left;
      height: $icon-height;
      width: $icon-width;
    }
  }

  .exitPopup {
    @extend %exit-popup;
  }

  .buttons {
    @include align-items(flex-end);
    display: flex;
    justify-content: space-between;
  }

  .scrollingArea {
    @extend %scrollbar;
    height: 385px;
    margin-top: 40px;

    .resourcesWrapper {
      display: flex;
      flex-wrap: wrap;
      padding: 0 30px;
    }
  }
}
