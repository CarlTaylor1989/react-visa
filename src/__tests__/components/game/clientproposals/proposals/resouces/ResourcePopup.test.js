/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import ResourcePopup from
  '../../../../../../components/game/clientproposals/proposals/resources/ResourcePopup';

let mockProps;
beforeEach(() => {
  mockProps = {
    showPopup: true,
    client: 'client1',
    closePopup: jest.fn(),
    type: '',
    resources: {
      0: {
        id: 'res1',
        name: 'RESOURCE NAME',
        text: 'TEXT',
        type: 'pdf',
        url: 'url'
      }
    },
    resourceId: 'res1'
  };
});

test('renders a hidden ResourcePopup component', () => {
  const wrapper = shallow(<ResourcePopup {...mockProps} showPopup={false} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});

test('renders a displayed ResourcePopup component', () => {
  const wrapper = shallow(<ResourcePopup {...mockProps} />);
  expect(wrapper.find('ResourceIcon').exists()).toBe(true);
});
