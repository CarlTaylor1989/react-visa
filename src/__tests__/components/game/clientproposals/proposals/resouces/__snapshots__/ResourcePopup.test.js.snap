// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a hidden ResourcePopup component 1`] = `
<div
  className="resourcePopup"
>
  <div
    className="popupWrapper"
  >
    <div
      className="buttons"
    >
      <ResourceIcon
        type=""
      />
      <button
        className="exitPopup"
        onClick={[MockFunction]}
        type="button"
      >
        <Translate
          tag="span"
          value="generic.popupCloseBtn"
        />
      </button>
    </div>
    <div
      className="scrollingArea"
    >
      <Scrollbar
        contentProps={Object {}}
        fallbackScrollbarWidth={20}
        maximalThumbYSize={83}
        minimalThumbSize={30}
        minimalThumbYSize={83}
        momentum={true}
        noDefaultStyles={true}
        scrollDetectionThreshold={100}
        scrollerProps={Object {}}
        thumbXProps={Object {}}
        thumbYProps={Object {}}
        trackClickBehavior="jump"
        trackXProps={Object {}}
        trackYProps={Object {}}
        wrapperProps={Object {}}
      >
        <ResouceContent
          client="client1"
          resource={
            Object {
              "id": "res1",
              "name": "RESOURCE NAME",
              "text": "TEXT",
              "type": "pdf",
              "url": "url",
            }
          }
          resourceIndex="0"
          type=""
        />
      </Scrollbar>
    </div>
  </div>
</div>
`;
